AssemblerLib/MeshComponentMap.cpp

```diff
@@ -285,4 +285,39 @@ std::vector<GlobalIndexType> MeshComponentMap::getGlobalIndicesByComponent(
     return global_indices;
 }
 
+GlobalIndexType MeshComponentMap::getLocalIndex(
+    Location const& l,
+    std::size_t const comp_id,
+    std::size_t const range_begin,
+    std::size_t const range_end) const
+{
+    GlobalIndexType const global_index = getGlobalIndex(l, comp_id);
+#ifndef USE_PETSC
+    return global_index;
+#else
+    if (global_index >= 0)    // non-ghost location.
+        return global_index - range_begin;
+
+    //
+    // For a ghost location look up the global index in ghost indices.
+    //
+    GlobalIndexType index;
+
+    // A special case for a ghost location with global index equal to the size
+    // of the local vector:
+    if (-global_index == _num_global_dof) return 0;
+
+    auto const ghost_index_it = std::find(_ghosts_indices.begin(),
```

Comment:
For the case of ``ComponentOrder::BY_COMPONENT`` and the number of components being one , the indices of entries of the local array are identical to the  indices of the nodes of the local partitioned mesh. For such case, the fetched local array from the global PETSc vector via the member of PETScVector can be directly used for output.  If `ComponentOrder::BY_COMPONENT`` and the number of components being large than one, to fetch a solution array of a component from the local solution array is also easy. 

For the case of ``ComponentOrder::BY_LOCATION``, the index mapping is not so straightforward. 

@endJunction I will prepare a PR to your PR.